lib/lib-table-cluster/src/main/java/org/sagebionetworks/table/cluster/TableIndexDAOImpl.java

```diff
@@ -466,14 +466,16 @@ public void createTableIfDoesNotExist(IdAndVersion tableId, boolean isView) {
 
 	@Override
 	public boolean alterTableAsNeeded(IdAndVersion tableId, List<ColumnChangeDetails> changes, boolean alterTemp) {
-		String sql = SQLUtils.createAlterTableSql(changes, tableId, alterTemp);
-		if(sql == null){
-			// no change are needed.
+		// record the number of columns in the table first
+		String countColumnsSql = SQLUtils.createCountNumberOfColumnsSql(tableId, alterTemp);
+		Long numColumns = template.query(countColumnsSql, new SingleColumnRowMapper<Long>()).get(0);

```

Comment:
the number of column budget is much higher than what we enforce, so we can skip the check on the number as n+1 is not going to trigger MySQL.